plugins/obs-outputs/rtmp-stream.c

```diff
@@ -1050,19 +1049,10 @@ static int try_connect(struct rtmp_stream *stream)
 
 	info("Connecting to RTMP URL %s...", stream->path.array);
 
-	// on reconnect we need to reset the internal variables of librtmp
-	// otherwise the data sent/received will not parse correctly on the other end
-	RTMP_Reset(&stream->rtmp);
-
-	// apparently TLS will not properly persist through connections
+	// free any existing RTMP TLS context
 	RTMP_TLS_Free(&stream->rtmp);
-	RTMP_TLS_Init(&stream->rtmp);
 
-	// since we don't call RTMP_Init above, there's no other good place
-	// to reset this as doing it in RTMP_Close breaks the ugly RTMP
-	// authentication system
-	memset(&stream->rtmp.Link, 0, sizeof(stream->rtmp.Link));
-	stream->rtmp.last_error_code = 0;
+	RTMP_Init(&stream->rtmp);
```

Comment:
I'm still worried this breaks something as there was a reason for the partial reset code we were doing before. I am almost certain it was for RTMP authentication, but as long as that still works we're probably fine.

Reply:
I spent hours looking through commits in librtmp to determine that the iterations of partial reset code were just the result of misunderstandings. The actual RTMP object should not be used at all across connections.